apps/communications/dialer/test/marionette/keypad_test.js

```diff
@@ -102,231 +105,270 @@ marionette('Dialer > Keypad', function() {
+
+      var number = subject.client.findElement(selectors.phoneNumber);
+      var args = [number, delPosition];
+      client.executeScript(function (numberElement, delPosition) {
+        numberElement.selectionStart = numberElement.selectionEnd = delPosition;
+        numberElement.click();
+      }, args);
+
+      var del = subject.client.findElement(selectors.del);
+      actions.tap(del).perform();
+      client.waitFor(function() {
+        return (number.getAttribute('value') == expectedNumberAfterDeletion);
+      });
+      assert.ok(true);
     });
-    assert.ok(true);
-  });
 
-  test('Deleting a couple of digits in the middle of the number', function() {
-    typeNumber();
+    test('Deleting a couple of digits in the middle of the number', function() {
+      typeNumber();
+
+      var number = subject.client.findElement(selectors.phoneNumber);
+      number.scriptWith(function (numberElement) {
+        numberElement.selectionStart = numberElement.selectionEnd = 2;
+        numberElement.click();
+      });
 
-    var number = subject.client.findElement(selectors.phoneNumber);
-    number.scriptWith(function (numberElement) {
-      numberElement.selectionStart = numberElement.selectionEnd = 2;
-      numberElement.click();
+      var del = subject.client.findElement(selectors.del);
+      actions.doubleTap(del).perform();
+
+      assert.equal(number.getAttribute('value'), '3');
     });
 
-    var del = subject.client.findElement(selectors.del);
-    actions.doubleTap(del).perform();
+    test('Clearing the number partially by long pressing the delete key',
+    function() {
+      typeNumber();
 
-    assert.equal(number.getAttribute('value'), '3');
-  });
+      var number = subject.client.findElement(selectors.phoneNumber);
+      number.scriptWith(function (numberElement) {
+        numberElement.selectionStart = numberElement.selectionEnd = 2;
+        numberElement.click();
+      });
 
-  test('Clearing the number partially by long pressing the delete key',
-  function() {
-    typeNumber();
+      var del = subject.client.findElement(selectors.del);
+      actions.longPress(del, 1).perform();
 
-    var number = subject.client.findElement(selectors.phoneNumber);
-    number.scriptWith(function (numberElement) {
-      numberElement.selectionStart = numberElement.selectionEnd = 2;
-      numberElement.click();
+      client.waitFor(function() {
+        return (number.getAttribute('value') === '3');
+      });
+      assert.ok(true, 'cleaned the phone number view');
     });
+});
 
-    var del = subject.client.findElement(selectors.del);
-    actions.longPress(del, 1).perform();
+suite('Add dialed number to a contact', function() {
+    function addTestContact(name, telephone) {
+    contacts.addContact({
+      givenName: name,
+      tel: telephone
+    });
+    }
 
-    client.waitFor(function() {
-      return (number.getAttribute('value') === '3');
+    setup(function() {
+      subject = new Dialer(client);
```

Comment:
TL;DR: I'd use `dialer` instead of `subject`.

I know it's been used at some other places in the tests, and I'm fine if we don't change it.

However, `subject` is not a self-explanatory name. You have to do an indirection (that is to say, either remember, or scroll back where it's defined, or take a look at the file path) to remember that's the dialer app you're testing. This can be confusing if you read many tests at the same time. For example, if somebody breaks many tests in many apps, because he/she changed something in `/shared`.